geomstats/geometry/special_orthogonal.py

```diff
@@ -1365,27 +1361,23 @@ def compose(self, point_1, point_2, point_type=None):
         if point_type is None:
             point_type = self.default_point_type
 
-        point_1 = self.regularize(point_1, point_type=point_type)
-        point_2 = self.regularize(point_2, point_type=point_type)
+        point_a = self.regularize(point_a, point_type=point_type)
+        point_b = self.regularize(point_b, point_type=point_type)
 
         if point_type == 'vector':
-            point_1 = self.matrix_from_rotation_vector(point_1)
-            point_2 = self.matrix_from_rotation_vector(point_2)
-
-        n_points_1 = point_1.shape[0]
-        n_points_2 = point_2.shape[0]
+            point_a = self.matrix_from_rotation_vector(point_a)
+            point_b = self.matrix_from_rotation_vector(point_b)
 
-        assert (point_1.shape == point_2.shape
-                or n_points_1 == 1
-                or n_points_2 == 1)
+        n_points_a = point_a.shape[0]
+        n_points_b = point_b.shape[0]
 
-        if n_points_1 == 1:
-            point_1 = gs.stack([point_1[0]] * n_points_2)
+        if n_points_a == 1:
```

Comment:
same as in SpecialEuclideand, this is not necessary. However there is an issue with einsum: numpy einsum does this well
```
point_a = rotations.random_uniform(4)
point_b = rotations.random_uniform(1)[0]
rot_mat_a = rotations.matrix_from_rotation_vector(point_a)
rot_mat_b = rotations.matrix_from_rotation_vector(point_b)[0] #2d-array
composition_rot_mat = np.einsum('...ij,...jk->...ik', rot_mat_a, rot_mat_b)
```
while gs.einsum throws `ValueError: Shape mismatch for einsum.`

Reply:
Great, thanks! Yes, this ValueError is one I have implemented because the new versions of einsum for tf and pytorch are not done yet. Working on it!